numpy/core/_add_newdocs.py

```diff
@@ -5491,6 +5491,39 @@
 
     """))
 
+add_newdoc('numpy.core.multiarray', 'dtype', ('metadata',
+    """
+    Either ``None`` or a readonly dictionary of metadata (mappingproxy).
+
+    The metadata field can be set using any dictionary at data-type
+    creation. Note that whether or not operations on arrays with metadata
+    attached to their datatypes is currently not well defined and should
+    not be relied on.
+
+    Examples
+    --------
+
+    >>> dt = np.dtype(float, metadata={"key": "value"})
+    >>> dt.metadata["key"]
+    'value'
+    >>> arr = np.array([1, 2, 3], dtype=dt)
+    >>> arr.dtype.metadata
+    mappingproxy({'key': 'value'})
+
+    Some operations may preserve metadata (identical data types):
+
+    >>> (arr + arr).dtype.metadata
+    mappingproxy({'key': 'value'})
+
+    But for example, adding two arrays with different metadata does not
+    propagate either one:
```

Comment:
```suggestion
    But if the metadata is different, the sum deletes both:
```

Reply:
"delete" conveys the wrong meaning - the original metadata is still there on the input arrays, it's just not propagated to the output.

Reply:
"Sum" doesn't mean addend or augend. "Sum" is a third thing, the output.

Reply:
I'd argue sum can also be the operation itself. At any rate, no deletion occurred - the output never had the metadata in the first place.

Reply:
> sum can also be the operation itself...no deletion occurred

Yes, those are good points.  

```suggestion
    But if the metadata is different, the sum propagates neither:
```

Reply:
Maybe just: for example the sum propagates neither?

One thing is that we drop metadata in such mixed operations most, but probably not all of the time (strings should be the biggest oddity, but there are few things you can do with strings).